### PR TITLE
ExceptionHandler, fix npe.

### LINKED ISSUE #20461: bleed, ExceptionHandler, npe
bleed, Ubuntu,

At start up. Due to a write permission issue an exception is correctly raised. However the handling of the exception causes below stack trace - and does not show the actual exception.

```
 ./launch-game.sh Game.Mod=ra
Platform is Linux
Engine version is {DEV_VERSION}
Runtime: .NET CLR 6.0.10
Exception of type `System.ArgumentNullException`: Value cannot be null. (Parameter 'format')
   at System.Text.StringBuilder.AppendFormatHelper(IFormatProvider provider, String format, ParamsArray args)
   at System.Text.StringBuilder.AppendFormat(String format, Object[] args)
   at OpenRA.ExceptionHandler.AppendIndentedFormatLine(StringBuilder sb, Int32 indent, String format, Object[] args) in /home/user/source/cs/OpenRA/OpenRA.Game/Support/ExceptionHandler.cs:line 61
   at OpenRA.ExceptionHandler.BuildExceptionReport(Exception ex, StringBuilder sb, Int32 indent) in /home/user/source/cs/OpenRA/OpenRA.Game/Support/ExceptionHandler.cs:line 102
   at OpenRA.ExceptionHandler.BuildExceptionReport(Exception ex, StringBuilder sb, Int32 indent) in /home/user/source/cs/OpenRA/OpenRA.Game/Support/ExceptionHandler.cs:line 99
   at OpenRA.ExceptionHandler.BuildExceptionReport(Exception ex) in /home/user/source/cs/OpenRA/OpenRA.Game/Support/ExceptionHandler.cs:line 56
   at OpenRA.ExceptionHandler.HandleFatalError(Exception ex) in /home/user/source/cs/OpenRA/OpenRA.Game/Support/ExceptionHandler.cs:line 49
   at OpenRA.Launcher.Program.Main(String[] args) in /home/user/source/cs/OpenRA/OpenRA.Launcher/Program.cs:line 36

Unhandled exception. System.ArgumentNullException: Value cannot be null. (Parameter 'format')
   at System.Text.StringBuilder.AppendFormatHelper(IFormatProvider provider, String format, ParamsArray args)
   at System.Text.StringBuilder.AppendFormat(String format, Object[] args)
   at OpenRA.ExceptionHandler.AppendIndentedFormatLine(StringBuilder sb, Int32 indent, String format, Object[] args) in /home/user/source/cs/OpenRA/OpenRA.Game/Support/ExceptionHandler.cs:line 61
   at OpenRA.ExceptionHandler.BuildExceptionReport(Exception ex, StringBuilder sb, Int32 indent) in /home/user/source/cs/OpenRA/OpenRA.Game/Support/ExceptionHandler.cs:line 102
   at OpenRA.ExceptionHandler.BuildExceptionReport(Exception ex, StringBuilder sb, Int32 indent) in /home/user/source/cs/OpenRA/OpenRA.Game/Support/ExceptionHandler.cs:line 99
   at OpenRA.ExceptionHandler.BuildExceptionReport(Exception ex) in /home/user/source/cs/OpenRA/OpenRA.Game/Support/ExceptionHandler.cs:line 56
   at OpenRA.ExceptionHandler.HandleFatalError(Exception ex) in /home/user/source/cs/OpenRA/OpenRA.Game/Support/ExceptionHandler.cs:line 49
   at OpenRA.Launcher.Program.Main(String[] args) in /home/user/source/cs/OpenRA/OpenRA.Launcher/Program.cs:line 36
Aborted (core dumped)
OpenRA has encountered a fatal error.
Please refer to the crash logs and FAQ for more information.

Log files are located in /home/user/.openra/Logs
The FAQ is available at http://wiki.openra.net/FAQ
````

After adding some logging before calling ExceptionHandler:

```
--------------begin of System.Console.WriteLine(e)
--------- System.UnauthorizedAccessException: Access to the path '/home/user/.openra/Logs/perf.log' is denied.
 ---> System.IO.IOException: Permission denied
   --- End of inner exception stack trace ---
   at Interop.ThrowExceptionForIoErrno(ErrorInfo errorInfo, String path, Boolean isDirectory, Func`2 errorRewriter)
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String path, OpenFlags flags, Int32 mode)
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at System.IO.Strategies.OSFileStreamStrategy..ctor(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize)
   at System.IO.Strategies.FileStreamHelpers.ChooseStrategy(FileStream fileStream, String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, Int64 preallocationSize)
   at System.IO.StreamWriter.ValidateArgsAndOpenPath(String path, Boolean append, Encoding encoding, Int32 bufferSize)
   at System.IO.File.CreateText(String path)
   at OpenRA.Log.AddChannel(String channelName, String baseFilename, Boolean isTimestamped) in /home/user/source/cs/OpenRA/OpenRA.Game/Support/Log.cs:line 148
   at OpenRA.Game.Initialize(Arguments args) in /home/user/source/cs/OpenRA/OpenRA.Game/Game.cs:line 348
   at OpenRA.Game.InitializeAndRun(String[] args) in /home/user/source/cs/OpenRA/OpenRA.Game/Game.cs:line 304
   at OpenRA.Launcher.Program.Main(String[] args) in /home/user/source/cs/OpenRA/OpenRA.Launcher/Program.cs:line 32
Exception of type `System.ArgumentNullException`: Value cannot be null. (Parameter 'format')
   at System.Text.StringBuilder.AppendFormatHelper(IFormatProvider provider, String format, ParamsArray args)
   at System.Text.StringBuilder.AppendFormat(String format, Object[] args)
   at OpenRA.ExceptionHandler.AppendIndentedFormatLine(StringBuilder sb, Int32 indent, String format, Object[] args) in /home/user/source/cs/OpenRA/OpenRA.Game/Support/ExceptionHandler.cs:line 61
   at OpenRA.ExceptionHandler.BuildExceptionReport(Exception ex, StringBuilder sb, Int32 indent) in /home/user/source/cs/OpenRA/OpenRA.Game/Support/ExceptionHandler.cs:line 102
   at OpenRA.ExceptionHandler.BuildExceptionReport(Exception ex, StringBuilder sb, Int32 indent) in /home/user/source/cs/OpenRA/OpenRA.Game/Support/ExceptionHandler.cs:line 99
   at OpenRA.ExceptionHandler.BuildExceptionReport(Exception ex) in /home/user/source/cs/OpenRA/OpenRA.Game/Support/ExceptionHandler.cs:line 56
   at OpenRA.ExceptionHandler.HandleFatalError(Exception ex) in /home/user/source/cs/OpenRA/OpenRA.Game/Support/ExceptionHandler.cs:line 49
   at OpenRA.Launcher.Program.Main(String[] args) in /home/user/source/cs/OpenRA/OpenRA.Launcher/Program.cs:line 37
--------------end of System.Console.WriteLine(e)

Unhandled exception. System.ArgumentNullException: Value cannot be null. (Parameter 'format')
   at System.Text.StringBuilder.AppendFormatHelper(IFormatProvider provider, String format, ParamsArray args)
   at System.Text.StringBuilder.AppendFormat(String format, Object[] args)
   at OpenRA.ExceptionHandler.AppendIndentedFormatLine(StringBuilder sb, Int32 indent, String format, Object[] args) in /home/user/source/cs/OpenRA/OpenRA.Game/Support/ExceptionHandler.cs:line 61
   at OpenRA.ExceptionHandler.BuildExceptionReport(Exception ex, StringBuilder sb, Int32 indent) in /home/user/source/cs/OpenRA/OpenRA.Game/Support/ExceptionHandler.cs:line 102
   at OpenRA.ExceptionHandler.BuildExceptionReport(Exception ex, StringBuilder sb, Int32 indent) in /home/user/source/cs/OpenRA/OpenRA.Game/Support/ExceptionHandler.cs:line 99
   at OpenRA.ExceptionHandler.BuildExceptionReport(Exception ex) in /home/user/source/cs/OpenRA/OpenRA.Game/Support/ExceptionHandler.cs:line 56
   at OpenRA.ExceptionHandler.HandleFatalError(Exception ex) in /home/user/source/cs/OpenRA/OpenRA.Game/Support/ExceptionHandler.cs:line 49
   at OpenRA.Launcher.Program.Main(String[] args) in /home/user/source/cs/OpenRA/OpenRA.Launcher/Program.cs:line 37
Aborted (core dumped)
```

### PR DESCRIPTION
**Pull request by anvilvapre**
Closes #20461.

### CODE REVIEW

**Review by RoosterDragon — DISMISSED** [review]

File: OpenRA.Game/Support/ExceptionHandler.cs [hunk 1]
```diff
@@ -56,9 +56,9 @@ static StringBuilder BuildExceptionReport(Exception ex)
 			return BuildExceptionReport(ex, new StringBuilder(), 0);
 		}
 
-		static StringBuilder AppendIndentedFormatLine(this StringBuilder sb, int indent, string format, params object[] args)
+		static StringBuilder AppendIndentedFormatLine(this StringBuilder sb, int indent, string message)
```
Comment (ThomasChr) [inline OpenRA.Game/Support/ExceptionHandler.cs]: After removing the format parameter the method name AppendIndentedFORMATLine seems rather misleading.

Comment (anvilvapre) [reply OpenRA.Game/Support/ExceptionHandler.cs]: Valid point. Remove Format?

Comment (ThomasChr) [reply OpenRA.Game/Support/ExceptionHandler.cs]: Seems like the right thing to do

**Review by ThomasChr — APPROVED** [review]

**Review by abcdefg30 — APPROVED** [review]

**Comment by abcdefg30** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog/_compare/d109dd2a90cc8c36f6f47d98e31b0d29bc84816d...b652c8edb2e3fc374aedaa634d7c8a4af259ff56)

https://github.com/OpenRA/OpenRA/commit/f70f2acb39334f3f1eaa0b484e33d074bdb93fa9

**Pull request merged**